packages/grid/x-data-grid/src/internals/utils/computeSlots.ts

```diff
@@ -18,5 +18,14 @@ export function computeSlots<SlotComponents extends object>({
     return defaultSlots;
   }
 
-  return { ...defaultSlots, ...overrides };
+  const result = { ...defaultSlots };
+  Object.keys(overrides).forEach((key) => {
+    const k = key as keyof typeof overrides;
+
+    if (overrides[k] !== undefined) {
+      result[k] = overrides[k] as any;
+    }
+  });
+
+  return result;
```

Comment:
thanks for the fix! little bit less boilerplatey suggestion if you want:

```suggestion
  return Object.fromEntries(Object.entries(defaultSlots).filter([k] => k !== undefined))
```

Reply:
Doesn't seem equivalent. And the result needs to be set to the default slot unless an override is provided.

Reply:
updated to be correct :)

Reply:
`Object.fromEntries(Object.entries(overrides).filter(fn)` is basically just a `pickBy` pattern, so if you all have lodash or a similar util you could reduce this code even a little bit further